src/main/java/org/mule/extensions/jms/internal/connection/provider/GenericConnectionProvider.java

```diff
@@ -93,4 +122,59 @@ public JmsSupport create102bSupport() {
       }
     };
   }
+
+  protected void configureSSLContextIfNeeded() {
+    if (!trustStorePassword.isEmpty() && !trustStoreName.isEmpty()) {
+      try {
+        final SSLContext context = SSLContext.getInstance(PROTOCOL);
+        context.init(new KeyManager[0],
+                     getCustomTrustStoreWithDefaultCerts(getTruststoreFile(trustStoreName), trustStorePassword),
+                     new SecureRandom());
+        SSLContext.setDefault(context);
+      } catch (Exception e) {
+        throw new JmsExtensionException("Failed to set custom TrustStore", e);
+      }
+    }
+  }
+
+  private TrustManager[] getCustomTrustStoreWithDefaultCerts(java.util.Optional<File> truststoreFile, String trustStorePassword)
+      throws Exception {
```

Comment:
Could we change the generic exception to a more specific one?

Reply:
done